File: app/src/main/java/com/nelyan/fragments/TradeFilterFragment.java
package com.nelyan.fragments;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.DatePicker;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.Spinner;
import android.widget.TextView;

import com.nelyan.R;
import com.nelyan.adapter.ActivityListAdapter;

import static com.nelyan.R.layout.customspinner;
import static com.nelyan.R.layout.size_customspinner;
import static com.nelyan.R.layout.trader_customspinner;

public class TradeFilterFragment extends Fragment implements AdapterView.OnItemSelectedListener, ActivityListAdapter.OnMyEventRecyclerViewItemClickListner {
    View v;
    Context mContext;
    private int mYear,mMonth,mDay;
    ImageView ivBack;
    TextView tvCal;
    Spinner orderby,orderby1;
    Dialog dialog;
Button btnFilter;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        v= inflater.inflate(R.layout.fragment_trade_filter, container, false); mContext = getActivity();
        ivBack=v.findViewById(R.id.ivBack);
        ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FragmentManager fm = getActivity().getSupportFragmentManager();
                Fragment f = fm.findFragmentById(R.id.frame_container);

                fm.popBackStack();
            }
        });
        LinearLayout ll_1=v.findViewById(R.id.ll_1);
        ll_1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dailogLocation();
            }
        });

        btnFilter=v.findViewById(R.id.btnFilter);
        btnFilter.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getActivity().onBackPressed();
            }
        });
     /*   tvCal=v.findViewById(R.id.tvCal);
        tvCal.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dateDialog();
            }
        });*/
        orderby=v.findViewById(R.id.orderby);
        final String[] genderlist = new String[]{
                "",
                "Sport","Cultural","Languages",

        };
        ArrayAdapter adapter = new ArrayAdapter(
                getActivity(), customspinner, genderlist);
        orderby.setAdapter(adapter);
        orderby.setOnItemSelectedListener(TradeFilterFragment.this);
        orderby1=v.findViewById(R.id.orderby1);
        final String[] km = new String[]{
              "",  "OKM","5KM","10KM","15KM","20KM",
                "25KM","30KM"

        };
        ArrayAdapter adapter1 = new ArrayAdapter(
                getActivity(), size_customspinner, km);
        orderby1.setAdapter(adapter1);
        orderby1.setOnItemSelectedListener(TradeFilterFragment.this);

        return v;
    }

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {

    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {

    }

    @Override
    public void onMyEventItemClickListner() {

    }
   /* private void dateDialog(){
        DatePickerDialog.OnDateSetListener listener= new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
                tvCal.setText(dayOfMonth + "/" + (monthOfYear+1)+"/"+year);
            }};
        DatePickerDialog datePickerDialog=new DatePickerDialog(getActivity(),R.style.datepicker,listener,mYear,mMonth,mDay);
        datePickerDialog.show();
    }*/
   public void dailogLocation(){

       dialog = new Dialog(mContext);
       dialog.getWindow().setBackgroundDrawableResource(android.R.color.transparent);
       dialog.setContentView(R.layout.alert_location);
       dialog.setCancelable(true);

       RelativeLayout rlYes,rlNo;
       rlNo= dialog.findViewById(R.id.rlNo);
       rlNo.setOnClickListener(new View.OnClickListener() {
           @Override
           public void onClick(View v) {
               //  mContext.startActivity(new Intent(mContext, HomeActivity.class));
               dialog.dismiss();
           }
       }); rlYes= dialog.findViewById(R.id.rlYes);
       rlYes.setOnClickListener(new View.OnClickListener() {
           @Override
           public void onClick(View v) {
               dialog.dismiss();
           }
       });
       dialog.show();
   }
}

File: app/src/main/java/com/nelyan_live/fragments/TradeFilterFragment.kt
/*
package com.nelyan_live.fragments

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import android.widget.AdapterView.OnItemSelectedListener
import androidx.fragment.app.Fragment
import com.nelyan_live.R
import com.nelyan_live.R.layout
import com.nelyan_live.adapter.ActivityListAdapter.OnMyEventRecyclerViewItemClickListner

class TradeFilterFragment : Fragment(), OnItemSelectedListener, OnMyEventRecyclerViewItemClickListner {
   lateinit  var v: View
    var mContext: Context? = null
    private val mYear = 0
    private val mMonth = 0
    private val mDay = 0
    var ivBack: ImageView? = null
    var tvCal: TextView? = null
    var orderby: Spinner? = null
    var orderby1: Spinner? = null
    var dialog: Dialog? = null
    var btnFilter: Button? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        v = inflater.inflate(layout.fragment_trade_filter, container, false)
        mContext = activity
        ivBack = v.findViewById(R.id.ivBack)
        ivBack!!.setOnClickListener(View.OnClickListener {
            val fm = requireActivity()!!.supportFragmentManager
            val f = fm.findFragmentById(R.id.frame_container)
            fm.popBackStack()
        })
        val ll_1 = v.findViewById<LinearLayout>(R.id.ll_1)
        ll_1.setOnClickListener { dailogLocation() }
        btnFilter = v.findViewById(R.id.btnFilter)
        btnFilter!!.setOnClickListener(View.OnClickListener { requireActivity()!!.onBackPressed() })
        */
/*   tvCal=v.findViewById(R.id.tvCal);
        tvCal.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dateDialog();
            }
        });*//*
orderby = v.findViewById(R.id.orderby)
        val genderlist = arrayOf<String?>(
                "",
                "Sport", "Cultural", "Languages")
        val adapter: ArrayAdapter<*> = ArrayAdapter<Any?>(
                requireActivity()!!, layout.customspinner, genderlist)
        orderby!!.setAdapter(adapter)
        orderby!!.setOnItemSelectedListener(this@TradeFilterFragment)
        orderby1 = v.findViewById(R.id.orderby1)
        val km = arrayOf<String?>(
                "", "OKM", "5KM", "10KM", "15KM", "20KM",
                "25KM", "30KM"
        )
        val adapter1: ArrayAdapter<*> = ArrayAdapter<Any?>(
                requireActivity()!!, layout.size_customspinner, km)
        orderby1!!.setAdapter(adapter1)
        orderby1!!.setOnItemSelectedListener(this@TradeFilterFragment)
        return v
    }

    override fun onItemSelected(parent: AdapterView<*>?, view: View, position: Int, id: Long) {}
    override fun onNothingSelected(parent: AdapterView<*>?) {}
    override fun onMyEventItemClickListner() {}

    */
/* private void dateDialog(){
        DatePickerDialog.OnDateSetListener listener= new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
                tvCal.setText(dayOfMonth + "/" + (monthOfYear+1)+"/"+year);
            }};
        DatePickerDialog datePickerDialog=new DatePickerDialog(getActivity(),R.style.datepicker,listener,mYear,mMonth,mDay);
        datePickerDialog.show();
    }*//*

    fun dailogLocation() {
        dialog = Dialog(mContext!!)
        dialog!!.window!!.setBackgroundDrawableResource(android.R.color.transparent)
        dialog!!.setContentView(layout.alert_location)
        dialog!!.setCancelable(true)
        val rlYes: RelativeLayout
        val rlNo: RelativeLayout
        rlNo = dialog!!.findViewById(R.id.rlNo)
        rlNo.setOnClickListener { //  mContext.startActivity(new Intent(mContext, HomeActivity.class));
            dialog!!.dismiss()
        }
        rlYes = dialog!!.findViewById(R.id.rlYes)
        rlYes.setOnClickListener { dialog!!.dismiss() }
        dialog!!.show()
    }
}*/
